### ISSUE TITLE
`xnu` not building on aarch64-darwin

### ISSUE THREAD

**Issue by kubukoz** [NixOS/nixpkgs#167683]
### Describe the bug

The xnu package, which is a dependency of openconnect via network_cmds, doesn't build on aarch64-darwin (M1 macs).

### Steps To Reproduce
1. Clone current master
2. `nix-build -A xnu`
3. See failure

### Expected behavior
Build succeeds

### Additional context

This might be related to #150880.

Failing build: https://hydra.nixos.org/build/172392835

```
clang-11: error: cannot use 'cpp-output' output with multiple -arch options
mig: fatal: "<no name yet>", line -1: no SubSystem declaration
```

### Notify maintainers

@NixOS/darwin-maintainers 

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"

- system: `"aarch64-darwin"`
 - host os: `Darwin 21.4.0, macOS 12.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.6.1`
 - channels(kubukoz): `"nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by toonn** [comment]
Did it build before the openssl drop?

I don't think any of the macOS 11 source releases build (except by coincidence) since the stdenv was based on the SDK because it's a lot of effort to get the source releases to build.

**Comment by kubukoz** [comment]
tbf it probably didn't build, sorry 😅 now I realised I was using the x86 versions of everything. These don't work now either, though - I'll open a new issue.

**Comment by willcohen** [comment]
See #170454 to fix network_cmds.

**Comment by jiegec** [comment]
See #179921 for a fix.

**Comment by jiegec** [comment]
Fixed by #179921.
